doc/specs/stdlib_bitsets.md

```diff
@@ -0,0 +1,2006 @@
+one of the error codes defined in this module. If absent, and its
+value would not have been `success`, then processing will stop with an
+informative text as its stop code. It shall have one of the error
+codes:
+
+* `success` - if no problems were found,
+
+* `alloc_fault` - if allocation of the bitset failed
+
+* `char_string_too_large_error` - if `string` was too large, or 
+
+* `char_string_invalid_error` - if string had an invalid character.
+
+
+#### Example
+
+```fortran
+    program demo_from_string
+        use stdlib_bitsets
+        character(*), parameter :: &
+            bits_all = '111111111111111111111111111111111'
+        type(bitset_64) :: set0
+        call set0 % from_string( bits_all )
+        if ( bits(set0) /= 33 ) then
+            error stop "FROM_STRING failed to interpret " // &
+                'BITS_ALL's size properly."
+        else if ( .not. set0 % all() ) then
+            error stop "FROM_STRING failed to interpret" // &
+                "BITS_ALL's value properly."
+        else
+            write(*,*) "FROM_STRING transferred BITS_ALL properly" // &
+                " into set0."
+        end if
+    end program demo_from_string
+```
+
+### `init` - `bitset_type` initialization routines.
+
+#### Status
+
+Experimental
+
+#### Description
+
+`bitset_type` initialization routine.
+
+#### Syntax
+
+`call [[stdlib_bitsets(module):init(interface)]] (self, bits [, status])`
+
+#### Class
+
+Subroutine.
+
+#### Arguments
+
+`self`: shall be a scalar `bitset_64` or `bitset_large` variable. It
+is an `intent(out)` argument.
+
+`bits` (optional): shall be a scalar integer expression of kind
+`bits_kind`. It is an `intent(in)` argument that if present
+specifies the number of bits in `set`. A negative value or a value
+greater than 64 if `self` is of type `bitset_64` is an error.
+
+`status` (optional): shall be a scalar default integer variable. It is
+an `intent(out)` argument that, if present, returns an error code
+indicating any problem found in processing `init`, and if absent and
+an error was found result in stopping processing with an informative
+stop code. It can have any of the following error codes:
+
+* `success` - no problem found
+
+* `alloc_fault` - `self` was of type `bitset_large` and memory
+  allocation failed
+
+* `array_size_invalid_error` - bits was present with either a negative
+  value, or a value greater than 64 when `self` was of type
+  `bitset_64`.
+
+#### Example
+
+```fortran
+    program demo_init
+        use stdlib_bitsets
+        type(bitset_large) :: set0
+        call set0 % init(166)
+        if ( set0 % bits() == 166 ) &
+            write(*,*) `SET0 has the proper size.'
+        if ( set0 % none() ) write(*,*) 'SET0 is properly initialized.'
+    end program demo_init
+```
+
+### `input` - reads a bitset from an unformatted file
+
+#### Status
+
+Experimental
+
+#### Description
+
+Reads a bitset from its binary representation in an unformatted
+file.
```

Comment:
Is it not worthwhile to mention the type of access used for unformatted files?

Reply:
I had been assuming sequential access, but it doesn't matter as long as the read is in the right position in the file. To ensure that you probably want to use the same access in the read as was used in the write.